command.go

```diff
@@ -0,0 +1,121 @@
+package githubactions
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// CommandProperties is a named "map[string]interface{}" type to hold key-value
+// pairs passed to an actions command.
+type CommandProperties map[string]interface{}
+
+// String encodes the CommandProperties to a string as comma separated
+// 'key=value' pairs. The pairs are joined in a chronological order.
+func (props *CommandProperties) String() string {
+	l := make([]string, 0, len(*props))
+	for k, v := range *props {
+		l = append(l, fmt.Sprintf("%s=%s", k, escapeProperty(v)))
+	}
+
+	sort.Strings(l)
+	return strings.Join(l, ",")
+}
+
+// Command can be issued by a GitHub action by writing to `stdout` with
+// following format.
+//
+// ::name key=value,key=value::message
+//
+//  Examples:
+//    ::warning::This is the message
+//    ::set-env name=MY_VAR::some value
+type Command struct {
+	Name       string
+	Message    interface{}
+	Properties CommandProperties
+}
+
+// String encodes the Command to a string in the following format:
+//
+// ::name key=value,key=value::message
+func (cmd *Command) String() string {
+	// https://github.com/actions/toolkit/blob/9ad01e4fd30025e8858650d38e95cfe9193a3222/packages/core/src/command.ts#L43-L45
+	if cmd.Name == "" {
+		cmd.Name = "missing.command"
+	}
+
+	const cmdSeparator = "::"
```

Comment:
Can we make this a global constant instead?

Reply:
Umm, sure! But I made it a local constant because it isn't needed anywhere else.

Reply:
Sure, but it gets defined as a constant each time the function is called. Making it global will save the allocs.

Reply:
oh! I thought the compiler would do a C style substitution. I have already made the change. will commit it with the rest of it.

Reply:
I don't think `const` gets reallocated on each function call: https://stackoverflow.com/a/38363661

Reply:
> The language spec itself makes no guarantees about such behavior

I think the compiler is just being smart, but we should be explicitly.

Reply:
Okay!